### PR TITLE
DAOS-13000 cart: Cleanup messages with hg returns

### PR DESCRIPTION
**Pull request by frostedcmos**
- Added DF_HG_RC/DP_HG_RC pair to print mercury error codes and the string version.
- Renamed crt_req_force_timeout -> crt_req_force_completion
- Changed crt_req_hg_send() to return void.

Required-githooks: true

Signed-off-by: Alexander Oganezov <alexander.a.oganezov@intel.com>
### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/cart/crt_context.c [hunk 1]
```diff
@@ -1880,7 +1880,7 @@ crt_context_set_timeout(crt_context_t crt_ctx, uint32_t timeout_sec)
 
 /* Execute handling for unreachable rpcs */
 void
-crt_req_force_timeout(struct crt_rpc_priv *rpc_priv)
+crt_req_force_completion(struct crt_rpc_priv *rpc_priv)
 {
 	struct crt_context	*crt_ctx;
```
Comment (ashleypittman) [inline src/cart/crt_context.c:1886]: The TRACE line below might also be wrong but that's minor.

**Review by ashleypittman — APPROVED** [review]
Thank you. Correct and verbose error reporting is important.

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 2]
```diff
@@ -390,6 +390,11 @@ def _check_pid_from_log_file(self, pid, abort_on_warning, leak_wf, show_memleaks
                             # than daos, so allow ENOMEM as well as
                             # -DER_NOMEM
                             show = False
+                        elif line.get_msg().endswith(': 5 (HG_NO_MEM)'):
+                            # Merury uses hg error numbers, rather
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_logtest.py:394]:
```suggestion
                            # Mercury uses hg error numbers, rather
```

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM except pylint/spelling

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 2 above)
Comment (frostedcmos) [reply src/tests/ftest/cart/util/cart_logtest.py]: fixed:)

**Review by daltonbohning — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Comment by ashleypittman** [comment]
Has a conflict with master, probababy from a PR of mine where I resolved one of these issues myself.

**Review by soumagne — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Review by soumagne — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Pull request merged**